### ISSUE TITLE
`polaris_viz_tooltip_root` position issue

### ISSUE THREAD

**Issue by Zwyx** [Shopify/polaris-viz#1596]
## Bug summary

Hi, I'm using a `LineChart` in the Shopify App Remix template, and the chart's tooltip isn't placed correctly:

![image](https://github.com/Shopify/polaris-viz/assets/29386932/4d6b4cc4-637c-43c1-bf07-46d565a1a525)

## Expected behavior

The tooltip should be on the chart.

## Actual behavior

The tooltip is on the bottom left of the iframe, cut in half.

## Steps to reproduce the problem

1. Add Polaris Viz to an app made with the Shopify App Remix template (which isn't trivial, see [here](https://github.com/Shopify/polaris-viz/issues/1425#issuecomment-1740896758))
1. Add a `LineChart`

## Specifications

- Polaris-Viz version number: `9.13.0`
- Browser: `Chrome 117`
- Device: `Laptop`
- Operating System: `Ubuntu 22.04`

**Comment by Zwyx** [comment]
Hey @paambaati, glad you also got Polaris Viz working in your Remix app :wink:  (over there: https://github.com/Shopify/polaris-viz/issues/1425#issuecomment-1742416921).

I'd be interested to know if you're also experiencing this issue with the chart tooltip?

**Comment by Zwyx** [comment]
No worries, thanks for replying!

**Comment by Darshan562** [comment]
> Hey @paambaati, glad you also got Polaris Viz working in your Remix app 😉 (over there: [#1425 (comment)](https://github.com/Shopify/polaris-viz/issues/1425#issuecomment-1742416921)).
> 
> I'd be interested to know if you're also experiencing this issue with the chart tooltip?

I face same issue when i use polaris-viz library in my shopify-remix app. then after i use rechrts for better ui.

![image](https://github.com/Shopify/polaris-viz/assets/96987506/7a7cb4eb-b2ab-42a1-a796-3b1ec4a32b02)

**Comment by HarshaKDeltaX** [comment]
Hey @Zwyx, Were you able to solve the issue with the tooltip? Even I'm facing the same issue with the tooltip position.

**Comment by Zwyx** [comment]
No, so I stopped using Polaris Viz. I use [Tremor](https://www.tremor.so/) now.

**Comment by ajayelsner** [comment]
If you guys are still facing the same problem just use the below solution to resolve your problem.
Just copy/paste the below CSS in your application:

div#polaris_viz_tooltip_root {
    position: fixed;
    width: auto;
    z-index: 99999999999999999;
    top: 0;
}
div#polaris_viz_tooltip_root ._Title_gk0ma_1,div#polaris_viz_tooltip_root ._Series_1mcj1_1{
    padding: 5px;
}

div#polaris_viz_tooltip_root ._Container_no9uw_1{
    max-width:100% !important;
}

**Comment by dochithanh2002** [comment]
Hi @ajayelsner 
Can you explain more about how you do that ?
Thank you so much.

**Comment by ComicDansMS** [comment]
> Hi @ajayelsner Can you explain more about how you do that ? Thank you so much.

What @ajayelsner has done most likely won't for others. The selector benig used is either randomly generated (not entirely sure how the build process works) or only for a particular component that matches the numbers and letters of that class name.

This library is kind of unusable as it currently stands (at least in the remix app template). Hugely frustrating as I want to build with the polaris design system as Shopify recommend.

**Comment by dochithanh2002** [comment]
Hi everyone, I found the solution for this issue!

This probliem occured basically when remix template is applied with polaris-viz library. The following line will not work using remix:
`import '@shopify/polaris-viz/build/esm/styles.css';`
This is why the css cannot be applied and render inside the charts. To fix this, we have to import css through `link`
Assume you want to create a chart in `app.check.js`, the following approach may be used:

In `app.check.js`
`import styles from "@shopify/polaris-viz/build/esm/styles.css";
//
export const links = () => [{ rel: "stylesheet", href: styles }];`

Ref: https://remix.run/docs/en/main/styling/css

**Comment by ComicDansMS** [comment]
And just like that, all of our problems disappear! Thank's a lot @dochithanh2002. I can't believe we missed such a basic fix haha.

**Comment by ComicDansMS** [comment]
> I don't plan to be using this library again (I have a feeling it's being neglected by Shopify...)

@Zwyx I have mixed feelings about this too. Part of me wants to ditch it as I don't feel great about using it in production, but the other part of me wants to use the Polaris design system as pushed by Shopify for continuity sake across the Shopify admin. Graphs and analytics are such a huge part of Shopify, and justifying why merchants should keep your app installed, so it's surprising to see such neglect of a key component.

**Comment by ngohoanglong** [comment]
`<div style={{ position: "relative", width: "100%", }}>
      <PolarisVizProvider themes={{
        default: {
          chartContainer: {
            padding: '20px',
            minHeight: 300,

          },
        }
      }}>

        {ready && (
          <LineChart
            showLegend={false}
            theme="default"
            data={data}
            tooltipOptions={
              {
                titleFormatter: function titleFormatter(value: any) { return new Date(value).toLocaleDateString() },
              }
            }
            xAxisOptions={{
              labelFormatter: function formatLinearXAxisLabel(value: any) { return new Date(value).toLocaleDateString("en-CA", { month: "short", day: "numeric", year: "numeric" }) },
            }}
            yAxisOptions={{
            }}
          />
        )}
      </PolarisVizProvider>
      <div id="polaris_viz_tooltip_root" style={{
        position: 'fixed',
        top: 0,
        left: 0,
        pointerEvents: 'none',
        transition: 'opacity 0.2s ease-in-out',
        zIndex: 1000,
      }}></div>
    </div>`
    it worked for me
